source/appModules/lync.py

```diff
@@ -0,0 +1,51 @@
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""appModule for Microsoft Skype for business. """
+ 
+import ui
+from NVDAObjects.UIA import UIA
+import appModuleHandler
+
+class NetUIRicherLabel(UIA):
+	"""A label sometimes found within list items that can fire live region changes, such as for chat messages."""
+
+	def event_liveRegionChange(self):
+		# The base liveRegionChange event is not enough as Skype for Business concatinates recent chat messages from the same person within the same minute
+		# Therefore, specifically strip out the chat content and only report the most recent part added.
+		# The object's name contains the full message (I.e. person: content, timestamp)
+		# The object's value just contains the content
+		# We are only interested in person and content
+		# Normalize the usage of end-of-line characters (name and value seem to expose them differently, which would break comparison)
+		content=self.value.replace('\r','\n').strip()
+		fullText=self.name.replace('\r\n\r\n','\n')
+		contentLines=content.split('\n')
+		contentStartIndex=fullText.find(content)

```

Comment:
I think it might be handy to have some examples of what these fields might look like before the processing is done. Or perhaps unit tests?